ggircs-app/app/components/FuelManagement/CarbonTaxActFuelType.tsx

```diff
@@ -0,0 +1,202 @@
+import { Card, Col, Row, Table } from "react-bootstrap";
+import React from "react";
+import { GraphQLTaggedNode, useFragment, graphql } from "react-relay";
+import { CarbonTaxActFuelType_query$key } from "__generated__/CarbonTaxActFuelType_query.graphql";
+import { FuelSelectionComponent } from "./FuelSelectionComponent";
+import { FuelChargeRow } from "./FuelChargeRow";
+import CreateEditFuelChargeRow from "./CreateEditFuelChargeRow";
+import { useRouter } from "next/router";
+import Button from "@button-inc/bcgov-theme/Button";
+
+interface Props {
+  query: CarbonTaxActFuelType_query$key;
+  pageQuery: GraphQLTaggedNode;
+}
+
+export const CarbonTaxActFuelType: React.FC<Props> = ({ query, pageQuery }) => {
+  const { carbonTaxActFuelType, allCarbonTaxActFuelTypes } = useFragment(
+    graphql`
+      fragment CarbonTaxActFuelType_query on Query
+      @argumentDefinitions(carbonTaxActFuelTypeId: { type: "ID!" }) {
+        carbonTaxActFuelType(id: $carbonTaxActFuelTypeId) {
+          id
+          rowId
+          carbonTaxFuelType
+          ctaRateUnits
+          fuelChargesByCarbonTaxActFuelTypeId(
+            first: 1000
+            orderBy: START_DATE_ASC
+          )
+            @connection(
+              key: "CarbonTaxActFuelType_fuelChargesByCarbonTaxActFuelTypeId"
+            ) {
+            __id
+            edges {
+              node {
+                id
+                startDate
+                endDate
+                ...FuelChargeRow_fuelCharge
+              }
+            }
+          }
+        }
+        allCarbonTaxActFuelTypes: query {
+          allCarbonTaxActFuelTypes {
+            edges {
+              node {
+                id
+                carbonTaxFuelType
+              }
+            }
+          }
+        }
+      }
+    `,
+    query
+  );
+
+  const router = useRouter();
+
+  const validateRatePeriod = (date: string, id?: string) => {
```

Comment:
A function name describing what validation is taking place here would be better. e.g.
```suggestion
  const isDateInFuelTypeFuelCharges = (date: string, fuelChargeId?: string) => {
```
is this `id` the fuel charge id? It's hard to tell